packages/plugin-js-packages/src/lib/runner/index.ts

```diff
@@ -22,3 +36,65 @@ export async function createRunnerConfig(
     outputFile: RUNNER_OUTPUT_PATH,
   };
 }
+
+export async function executeRunner(): Promise<void> {
+  const { packageManager, checks, auditLevelMapping } =
+    await readJsonFile<FinalJSPackagesPluginConfig>(PLUGIN_CONFIG_PATH);
+
+  const auditResults = checks.includes('audit')
+    ? await processAudit(packageManager, auditLevelMapping)
+    : [];
+
+  const outdatedResults = checks.includes('outdated')
+    ? await processOutdated(packageManager)
+    : [];
+  const checkResults = [...auditResults, ...outdatedResults];
+
+  await ensureDirectoryExists(dirname(RUNNER_OUTPUT_PATH));
+  await writeFile(RUNNER_OUTPUT_PATH, JSON.stringify(checkResults));
+}
+
+async function processOutdated(packageManager: PackageManager) {
+  const { stdout } = await executeProcess({
+    command: packageManager,
+    args: ['outdated', '--json', '--long'],
+    alwaysResolve: true, // npm outdated returns exit code 1 when outdated dependencies are found
+  });
+
+  const outdatedResult = JSON.parse(stdout) as NpmOutdatedResultJson;
+  return packageDependencies.map(dep =>
+    outdatedResultToAuditOutput(outdatedResult, dep),
+  );
+}
+
+async function processAudit(
+  packageManager: PackageManager,
+  auditLevelMapping: Record<PackageAuditLevel, IssueSeverity>,
+) {
+  const auditResults = await Promise.allSettled(
+    packageDependencies.map<Promise<AuditOutput>>(async dep => {
+      const { stdout } = await executeProcess({
+        command: packageManager,
+        args: ['audit', ...getNpmAuditOptions(dep)],
+      });
+
+      const auditResult = JSON.parse(stdout) as NpmAuditResultJson;
+      return auditResultToAuditOutput(auditResult, dep, auditLevelMapping);
+    }),
+  );
+  return auditResults
+    .filter(
+      (x): x is PromiseFulfilledResult<AuditOutput> => x.status === 'fulfilled',
+    )
+    .map(x => x.value);
+}
```

Comment:
Looks like this silences rejected promises. Maybe at least print warnings to the console?

Reply:
Well, we currently do not print any plugin output in `core` unless it throws so nothing will be printed even if I do 😅